### PR TITLE
fix(v2): modify validation schema and tests for rehype/remark + remove duplicate dependency

### LINKED ISSUE #3125: Error when including rehype plugins
## 🐛 Bug Report

Unable to add rehypePlugins in docusaurus.config.js. Results in following error during `docuasaurus start` and `docusaurus build`.

`ValidationError: "rehypePlugins[0]" must be a string`

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes.

## To Reproduce

(Write your steps here:)

1. `npx @docusaurus/init@next init docusaurus classic`
2. `npm i --save remark-math rehype-katex`
3. In docusaurus.config.js import the two packages and add to docs:
```
const katex = require("rehype-katex");
const math = require("remark-math");

module.exports = {
  // ...
  presets: [
    [
      "@docusaurus/preset-classic",
      {
        docs: {
          // It is recommended to set document id as docs home page (`docs/` path).
          homePageId: "doc1",
          sidebarPath: require.resolve("./sidebars.js"),
          remarkPlugins: [math],
          rehypePlugins: [katex],
          // Please change this to your repo.
          editUrl:
            "https://github.com/facebook/docusaurus/edit/master/website/",
        },
        // ...
      },
    ],
  ],
};

```
4. `docusaurus start`

## Expected behavior

Expect to be able to serve website.

## Actual Behavior

```
docusaurus start

Starting the development server...
ValidationError: "rehypePlugins[0]" must be a string
```

(Write what happened. Add screenshots, if applicable.)

## Your Environment

- Docusaurus version used: 2.0.0-alpha.59
- Environment name and version (e.g. Chrome 78.0.3904.108, Node.js 10.17.0): Node 14.5.0
- Operating system and version (desktop or mobile): Windows 10

## Reproducible Demo

https://github.com/mattcorner/docusaurus-equations-error/blob/master/docusaurus.config.js

### PR DESCRIPTION
**Pull request by teikjun**
## Motivation

This PR makes some changes related to validation:
- modify validation schema of rehype/remark fields to accept object type. This fixes #3125. Previously, I assumed that remark/rehype plugins could only be function type (AKA [attachers](https://github.com/unifiedjs/unified#function-attacheroptions) in unified's lingo), but this is not exhaustive. It is also possible that rehype/remark plugins are object type.  Based on [unified's documentation](https://github.com/unifiedjs/unified#processoruseplugin-options) and [typescript types](https://github.com/unifiedjs/unified/blob/main/types/index.d.ts#L13), I believe that we have included all the possible types for rehype/remark plugins after adding object type.
- validate babel style user input
- add tests for the above
- remove a duplicate dependency for utils-validation

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan
Automated tests

### CODE REVIEW

File: packages/docusaurus-utils-validation/src/validationSchemas.ts [hunk 1]
```diff
@@ -14,10 +14,9 @@ export const PluginIdSchema = Joi.string()
...
   .items(
     Joi.array()
-      // TODO, this allows [config,fn] too?
-      .items(Joi.function().required(), Joi.object().required())
-      .length(2),
+      .ordered(Joi.function().required(), Joi.object().required()),
     Joi.function(),
+    Joi.object(),
   )
   .default([]);
```
Comment (teikjun) [inline packages/docusaurus-utils-validation/src/validationSchemas.ts:21]: This completes the TODO for validating 'babel style' input and fixes the validation issue

File: packages/docusaurus/package.json [hunk 2]
```diff
@@ -107,8 +107,7 @@
     "webpack-bundle-analyzer": "^3.6.1",
     "webpack-dev-server": "^3.11.0",
     "webpack-merge": "^4.2.2",
-    "webpackbar": "^4.0.0",
-    "@docusaurus/utils-validation": "^2.0.0-alpha.61"
```
Comment (teikjun) [inline packages/docusaurus/package.json:111]: This is a duplicate dependency, the other one is on line 50

File: packages/docusaurus-plugin-content-docs/src/__tests__/pluginOptionSchema.test.ts [hunk 3]
```diff
@@ -9,12 +9,14 @@ import {PluginOptionSchema, DEFAULT_OPTIONS} from '../pluginOptionSchema';
...
-const remarkRehypePluginStub = () => {};
+const markdownPluginsFunctionStub = () => {};
+const markdownPluginsObjectStub = {};
 
 describe('normalizeDocsPluginOptions', () => {
   test('should return default options for undefined user options', async () => {
-    const {value} = await PluginOptionSchema.validate({});
+    const {value, error} = await PluginOptionSchema.validate({});
     expect(value).toEqual(DEFAULT_OPTIONS);
+    expect(error).toBe(undefined);
```
Comment (teikjun) [inline packages/docusaurus-plugin-content-docs/src/__tests__/pluginOptionSchema.test.ts:19]: It is possible for value to match when there is an error, so it is safer to include this

File: packages/docusaurus-utils-validation/src/validationSchemas.ts (hunk 1 above)
Comment (anshulrgoyal) [reply packages/docusaurus-utils-validation/src/validationSchemas.ts]: ordered is really cool

Comment (slorber) [reply packages/docusaurus-utils-validation/src/validationSchemas.ts]: yes, looked for that but couldn't find it. Joi doc is a pain :D

**Comment by slorber** [comment]
LGTM thanks

**Pull request merged**